src/main/java/com/example/demo/serviceIml/StudentServiceImpl.java

```diff
@@ -19,101 +21,134 @@
+        }
+
+        // check if mail is valid
+        if (!Utils.isMailValid(student.getStudentEmail())) {
+            log.error(CommonResponses.emailNotValidMsg);
+            throw new BadRequestException(CommonResponses.emailNotValidMsg);
+        }
+
+        if (student.getStudentName().length() < 2) {
+            log.error(CommonResponses.nameNotValidMsg);
+            throw new BadRequestException(CommonResponses.nameNotValidMsg);
+        }
+
+        studentRepository.save(student);
+        log.info("New student saved: {}", student.toString());
+    }
+
+    @Transactional
+    @Override
+    public void updateStudent(int studentId, final Student updateStudent) {
+        // check if the student is exist
+        if (!studentRepository.existsById(studentId)) {
+            log.error(studentNotExistByIdMsg + studentId);
+            throw new NotFoundException(studentNotExistByIdMsg + studentId);
+        }
+
+        Student currentStudent = studentRepository.findById(studentId).get();
+
+        // Student name processing
+        if (updateStudent.getStudentName().length() < 2) {
+            log.error(CommonResponses.nameNotValidMsg);
+            throw new BadRequestException(CommonResponses.nameNotValidMsg);
+        }
+        currentStudent.setStudentName(updateStudent.getStudentName());
+
+        // Check student email if valid
+        if (!Utils.isMailValid(updateStudent.getStudentEmail())) {
+            log.error(CommonResponses.emailNotValidMsg);
+            throw new BadRequestException(CommonResponses.emailNotValidMsg);
+        }
+
+        // check if e-mail taken
+        if (studentRepository.isExistByEmail(updateStudent.getStudentEmail())) {
+            log.error(CommonResponses.emailTakenMsg);
+            throw new BadRequestException(CommonResponses.emailTakenMsg);
+        }
+        currentStudent.setStudentEmail(updateStudent.getStudentEmail());
+
+        // no need process for the DOB becase it's @NonNull in the entity class
+        currentStudent.setStudentDOB(updateStudent.getStudentDOB());
+        log.info("Student updated: {} ", currentStudent.toString());
+    }
+
+    @Override
+    public void deleteStudentById(int studentId) {
+        if (!studentRepository.existsById(studentId)) {
+            log.error(studentNotExistByIdMsg + studentId);
+            throw new NotFoundException(studentNotExistByIdMsg + studentId);
+        }
+        studentRepository.deleteById(studentId);
+        log.info(studentSuccessfullyDeleteMsg + studentId);
+    }
+
+    @Override
+    public void deleteStudentByEmail(String studentEmail) {
+        if (!studentRepository.isExistByEmail(studentEmail)) {
+            log.error(studentNotExistByEmailMsg + studentEmail);
+            throw new NotFoundException(studentNotExistByEmailMsg + studentEmail);
+        }
+        studentRepository.deleteByEmail(studentEmail);
+        log.info(studentSuccessfullyDeleteMsg + studentEmail);
+    }
```

Comment:
Here there is no student ID but the message is like this `Student successfully deleted with E-mail: ` Refactor message like below
`Student successfully deleted with ` is the root message. Add E-mail or ID in the function